addons/web/static/src/js/widgets/data_export.js

```diff
@@ -496,7 +496,19 @@ var DataExport = Dialog.extend({
                 import_compat: !!this.$import_compat_radios.filter(':checked').val(),
             })},
             complete: framework.unblockUI,
-            error: crash_manager.rpc_error.bind(crash_manager),
+            error: function (error) {
+                if (error.code === 504 && error.message === '504 Gateway Time-out') {
```

Comment:
Why not handle this in the crash manager directly, and why change the format of the error object completely when there's still a hard dependency on the message itself?

Reply:
@xmo-odoo  We already handled this in `crash_manager` but that is generic error message. We have to show specific error message when timeout happen during export large amount of data.

Reply:
@jat-odoo I've rebuilt the entire get_file function & handled this case in what I think is a cleaner way in #25731 using updated/modern browser APIs obviating the need for iframes. Assuming @ged-odoo is OK with merging that I'll just remove this change from here when merging it.